### PR TITLE
Feature/feature new pet collections

### PR DESCRIPTION
**Pull request by marclll**

### CODE REVIEW

File: usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/ExerciseData.java [hunk 1]
```diff
@@ -0,0 +1,147 @@
+package org.pesmypetcare.usermanagerlib.datacontainers;
+
+import android.webkit.HttpAuthHandler;
```
Comment (santidrj) [inline usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/ExerciseData.java:3]: Codacy Issue found: [Unused import - android.webkit.HttpAuthHandler.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5493777)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Coverage decreased per file
===========================
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/MedicationData.java  -18
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/Weight.java  -15
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/MealData.java  -10
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/WashData.java  -13
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/Meal.java  -15
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/WeightData.java  -13
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/Wash.java  -15
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/Medication.java  -32
         

Complexity increasing per file
==============================
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/Exercise.java  2
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/Vaccination.java  2
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/VetVisit.java  2
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/Illness.java  2
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/ExerciseData.java  3
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/IllnessData.java  2
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/VaccinationData.java  2
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/VetVisitData.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5493777&bid=18003803)

File: usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/ExerciseData.java [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+    public ExerciseData(String name, String description, String endDateTime, @NonNull List<LatLng> coordinates) {
+        PetData.checkDateFormat(endDateTime);
+        Map<String, Double> point;
+        this.name = name;
+        this.description = description;
+        this.endDateTime = endDateTime;
+        this.coordinates = new ArrayList<>();
+        for (LatLng lat: coordinates) {
+            if (lat != null) {
+                point = new HashMap<>();
```
Comment (santidrj) [inline usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/ExerciseData.java:58]: Codacy Issue found: [Avoid instantiating new objects inside loops](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5493777)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage decreased per file
===========================
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/WeightData.java  -13
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/Weight.java  -15
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/WashData.java  -13
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/Wash.java  -15
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/MedicationData.java  -18
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/Medication.java  -32
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/Meal.java  -15
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/MealData.java  -10
         

Complexity increasing per file
==============================
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/VetVisitData.java  2
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/Illness.java  2
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/VaccinationData.java  2
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/IllnessData.java  2
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/VetVisit.java  2
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/Exercise.java  2
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/ExerciseData.java  3
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/Vaccination.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5493777&bid=18003803)

File: usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/PetData.java [hunk 3]
```diff
@@ -226,7 +250,127 @@ public static void checkDateAndValueInteger(String key, Map<String, Object> body
...
+     * @param severity Severity value
+     */
+    private static void checkSeverityValue(String severity) {
+        if (!"Low".equals(severity) && !"Medium".equals(severity) && !"High".equals(severity)) {
+            throw new IllegalArgumentException("Incorrect severity format");
+        }
+    }
+
+    /**
+     * Checks wether type value is valid or not
```
Comment (AlbertPG) [inline usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/PetData.java:363]: It is missing a dot

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/VetVisit.java  2
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/VaccinationData.java  2
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/VetVisitData.java  2
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/IllnessData.java  2
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/Exercise.java  2
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/ExerciseData.java  3
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/Illness.java  2
- usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/Vaccination.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5493777&bid=18003803)

**Pull request merged**